src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/TestCenter.java

```diff
@@ -32,17 +30,19 @@ public TestCenter(BloxbergClient client, File outputfile) {
 
     public void generateJsonFile(int start, int stop) throws IOException {
 
-        FileWriter fileWriter = new FileWriter(outputfile, true);
+        //FileWriter fileWriter = new FileWriter(outputfile, true);
         ObjectMapper objectMapper = new ObjectMapper();
-        SequenceWriter seqWriter = objectMapper.writer().writeValuesAsArray(fileWriter);
+        ReducedTransObjectArray rdoa = new ReducedTransObjectArray();
+        //SequenceWriter seqWriter = objectMapper.writer().writeValuesAsArray(fileWriter);
         for (int i = start; start <= stop; start ++) {
-            extractJsonObject(start, seqWriter);
+            extractJsonObject(start, rdoa);
         }
-        seqWriter.close();;
+        //seqWriter.close();;
+        objectMapper.writeValue(outputfile, rdoa);
 
     }
 
-    public void extractJsonObject(int blockNumber, SequenceWriter seqWriter) throws IOException {
+    public void extractJsonObject(int blockNumber, ReducedTransObjectArray rdoa) throws IOException {
```

Comment:
Don't make a void method and then mutate a parameter. Instead, use return values.

Reply:
but it s still ok to mutate a parameter and use that parameter then as a return value, right?

Reply:
No, this is terrible style and not idiomatic Java. 
If you want to mutate values, mutate your fields (or create completely new objects). Follow OOP encapsulation principles.